regulations/check_missing.py

```diff
@@ -0,0 +1,34 @@
+import os
+import subprocess
+
+import re
+MAX_LANG_WIDTH = 20
+FNULL = open(os.devnull, 'w')
+WARNING_FORMAT = ('%s - %s does not match a regulation, '
+                  'please change "%s" to "[SEPARATE]%s".\n')
+
+def check_guidelines(language, base_path):
+  padded_language = ("[" + language + "]").ljust(MAX_LANG_WIDTH +2)
+  with open(base_path + "/wca-guidelines.md") as guidelines:
+    warnings = ""
+    for item in guidelines:
+      # Only match the lines starting with something like "- A1b++)" (guidelines),
+      # which don't have the "SEPARATE" label.
+      match_result = re.match(r"- (\w+)(\++)\) (?!\[SEPARATE\])(\[[^\[\]]+\])", item)
+      if match_result is None:
+        continue
+      # Extract the regulation we want to look for
+      # '0' is the full match, '1' is the first subgroup (regulation), '2' is the subnumber
+      regulation, subnumber, label = match_result.groups()
+
+      res = subprocess.call(["grep", r"\- " + regulation + ")", base_path + "/wca-regulations.md"],
```

Comment:
Yeah I must admit I hesitated between going full python and using grep.
What made me choose grep is that using python I would have to open the file, loop through all the lines and check if it matches, while grep can do this for me and I can just check its return code.
And also I would have to write an "intelligent" function that reuse the file handler to avoid opening it every time.

I can do that of course if you think it's worth it !

What do you think makes it harder to read ? Using a process return code instead of a match result ?